Lib/typing.py

```diff
@@ -2436,6 +2439,10 @@ def _overload_dummy(*args, **kwds):
         "by an implementation that is not @overload-ed.")
 
 
+# {module: {qualname: {firstlineno: func}}}
+_overload_registry = defaultdict(lambda: defaultdict(dict))
```

Comment:
Using `functools.partial` seems to be pretty consistently faster than a `lambda` on my machine:

```python
C:\Users\alexw\coding\cpython>python
Running Debug|x64 interpreter...
Python 3.11.0a7+ (main, Apr 14 2022, 10:41:31) [MSC v.1931 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from functools import partial
>>> from collections import defaultdict
>>> _jelle_overload_registry = defaultdict(lambda: defaultdict(dict))
>>> _alex_overload_registry = defaultdict(partial(defaultdict, dict))
>>> def _overload_dummy(*args, **kwds):
...     raise NotImplementedError("no")
>>> def overload_jelle(func):
...     f = getattr(func, "__func__", func)
...     try:
...         _jelle_overload_registry[f.__module__][f.__qualname__][f.__code__.co_firstlineno] = func
...     except AttributeError:
...         pass
...     return _overload_dummy
...
>>>  def overload_alex(func):
...     f = getattr(func, "__func__", func)
...     try:
...         _alex_overload_registry[f.__module__][f.__qualname__][f.__code__.co_firstlineno] = func
...     except AttributeError:
...         pass
...     return _overload_dummy
...
>>> def f(): ...
>>> import timeit
>>> timeit.timeit(lambda: overload_jelle(f), number=1000000)
1.296042200177908
>>> timeit.timeit(lambda: overload_alex(f), number=1000000)
1.2557714998256415
```

Reply:
I feel like the lambda variant is easier to read, but I guess every microsecond counts :)

Reply:
I don’t like partial either, but here it seems right.

Reply:
Yeah, agreed — the lambda is better for readability, but in this situation, we have a need for speed :)